### PR TITLE
Fix examples/generate_bom/generate_bom.py to work without pcb

### PR DESCRIPTION
**Pull request by kdumontnu**
Fixes `generate_bom.py` (which I'm using to test #83)

### CODE REVIEW

**Comment by shrik450** [comment]
I was waiting on updating this until I was done with variants, because each time the API for bom_generation will break and we'll need to update the script. E.g. with #84 there's more arguments which need to be parsed and evaluated. If you don't mind using the updated script for review, I'll make all the example changes in a separate PR after the three BOM PRs are done.

**Comment by kdumontnu** [comment]
> I was waiting on updating this until I was done with variants, because each time the API for bom_generation will break and we'll need to update the script. E.g. with #84 there's more arguments which need to be parsed and evaluated. If you don't mind using the updated script for review, I'll make all the example changes in a separate PR after the three BOM PRs are done.

Why wouldn't we change it as we go to keep it functional with the current version?

File: examples/generate_bom/generate_bom.py [hunk 1]
```diff
@@ -24,12 +24,10 @@
...
         prog="generate_bom", description="Generate a BOM from a PrjPcb file."
     )
-    parser.add_argument("repository", help="The repo containing the project")
-    parser.add_argument("prjpcb_file", help="The path to the PrjPcb file in the source repo.")
     parser.add_argument(
-        "pcb_file",
-        help="The path to the PCB file in the source repo.",
+        "repository", help="The repo containing the project in the form 'owner/repo'"
     )
+    parser.add_argument("prjpcb_file", help="The path to the PrjPcb file in the source repo.")
```
[the remark this answers is not in the document]
Comment (kdumontnu) [reply examples/generate_bom/generate_bom.py:30]: Don’t we still need the project file?

[the remark this answers is not in the document]
Comment (shrik450) [reply examples/generate_bom/generate_bom.py:30]: Ah sorry, I mixed this up with the pcbdoc file because of the red change above it.

**Pull request merged**